server/src/main/java/org/elasticsearch/plugins/PluginDescriptor.java

```diff
@@ -106,6 +108,9 @@ public PluginDescriptor(
         this.isLicensed = isLicensed;
         this.isModular = isModular;
         this.isStable = isStable;
+        this.builtWithSemanticVersion = Optional.ofNullable(elasticsearchVersion)
```

Comment:
I get what you are trying to do, but I don't think this is something we need to worry about right now. Plugins cannot be built outside of an ES "semantic version", even for serverless. I think we should change this simply here for now (just assume a semantic version string), and figure out the best way to represent serverless plugins later when we are ready for that.

Reply:
I don't know about "always", but for the time being, the build won't change (we will continue to have the semantic version in the build). Note this isn't just plugin metadata, we have the semantic version in the file paths themselves. This is ok since it is completely local to a node; we don't actually care what the filenames are.